reconciler/reconciler.go

```diff
@@ -331,6 +106,24 @@ func (r *Reconciler) wrappedInactiveEnqueue(
 	}
 }
 
+// addToqueueMap adds a *types.AccountCurrency
+// to the prune map at the provided index.
+func (r *Reconciler) addToqueueMap(
+	acctCurrency *types.AccountCurrency,
+	index int64,
+) {
+	r.queueMapMutex.Lock()
+	key := types.Hash(acctCurrency)
```

Comment:
Can we put this outsider of the lock?

```suggestion
	key := types.Hash(acctCurrency)
	r.queueMapMutex.Lock()
```